src/neptune/internal/utils/images.py

```diff
@@ -196,38 +200,51 @@ def _image_content_to_html(content: bytes) -> str:
     return "<img src='data:image/png;base64," + str_equivalent_image + "'/>"
 
 
-def _get_numpy_as_image(array):
+def _get_numpy_as_image(array: np.ndarray, autoscale: bool) -> bytes:
     array = array.copy()  # prevent original array from modifying
+    if autoscale:
+        array = _scale_array(array)
 
-    data_range_warnings = []
-    array_min = array.min()
-    array_max = array.max()
-    if array_min < 0:
-        data_range_warnings.append(f"the smallest value in the array is {array_min}")
-    if array_max > 1:
-        data_range_warnings.append(f"the largest value in the array is {array_max}")
-    if data_range_warnings:
-        data_range_warning_message = (" and ".join(data_range_warnings) + ".").capitalize()
-        logger.warning(
-            "%s To be interpreted as colors correctly values in the array need to be in the [0, 1] range.",
-            data_range_warning_message,
-        )
-    array *= 255
-    shape = array.shape
-    if len(shape) == 2:
+    if len(array.shape) == 2:
         return _get_pil_image_data(pilimage_fromarray(array.astype(numpy_uint8)))
-    if len(shape) == 3:
-        if shape[2] == 1:
+    if len(array.shape) == 3:
+        if array.shape[2] == 1:
             array2d = numpy_array([[col[0] for col in row] for row in array])
             return _get_pil_image_data(pilimage_fromarray(array2d.astype(numpy_uint8)))
-        if shape[2] in (3, 4):
+        if array.shape[2] in (3, 4):
             return _get_pil_image_data(pilimage_fromarray(array.astype(numpy_uint8)))
     raise ValueError(
         "Incorrect size of numpy.ndarray. Should be 2-dimensional or"
         "3-dimensional with 3rd dimension of size 1, 3 or 4."
     )
 
 
+def _scale_array(array: np.ndarray) -> np.ndarray:
+    array_min = array.min()
+    array_max = array.max()
+
+    if array_min >= 0 and 1 < array_max <= 255:
+        return array
+
+    if array_min >= 0 and array_max <= 1:
+        return array * 255
+
+    if array_min >= -1 and array_max <= 1:
+        return (array + 1) / 2 * 255
```

Comment:
This is a very unusual image representation. Maybe it is better not to handle this case and just warn about incorrect data range

Reply:
Yeah, I think that we actually need to have a discussion about what we want to support and how to proceed with exotic image representations and warnings :face_exhaling:

Reply:
These are normalized images in the range [-1,1].
This range is generally used as an input for models, not for rendering images.

Most users would be normalizing images, so IMO, we should take care of rendering it from our end

Reply:
@SiddhantSadangi 
Most of the input transformation in deep learning is obtained by [normalization](https://pytorch.org/vision/main/generated/torchvision.transforms.Normalize.html) which aims to make image channels to have mean 0 and std 1.

Usually process looks like this: scale to range [0, 1] -> apply normalization (might be out of [0, 1 range]) -> feed into network 

I would be very coucious when it comes to scaling images for user behind the user. It might be the case that normalized images fit into range [0, 1] and naive linear scaling will disturb the image (we need to know mean and std of dataset to perform inverse transformation) confusing end user.

I suggest picking the most common representations, ex. recognized by other libraries like matplotlib. Documenting them properly  and warning user when other conventions are detected

Reply:
I agree with @szysad and I want to be consistent with what matplotlib does. They're supporting `[0-255]` and `[0.0-1.0]` and there is a warning and clipping to `[0, 1]` in other cases like `[-1, 1]`